### PR TITLE
Agent entity annotations & new attributes

### PR DESCRIPTION
**Pull request by adamkral12**
added rank & languages

### CODE REVIEW

File: Secret-agency-persistence/src/main/java/cz/fi/muni/pa165/secretagency/entity/Agent.java [hunk 1]
```diff
@@ -1,34 +1,55 @@
...
+    @Temporal(TemporalType.DATE)
+    @NotNull
     private LocalDateTime birthDate;
 
+    @ElementCollection
+    @NotNull
+    private Set<LanguageEnum> languages = new HashSet<>();
+
+    @NotNull
+    private AgentRankEnum rank;
```
Comment (tremors0) [inline Secret-agency-persistence/src/main/java/cz/fi/muni/pa165/secretagency/entity/Agent.java]: Tady chybí anotace @Enumerated.

**Comment by adamkral12** [comment]
nn, kouknu na to vecer, dik

File: Secret-agency-persistence/src/main/java/cz/fi/muni/pa165/secretagency/entity/Agent.java [hunk 2]
```diff
@@ -1,34 +1,55 @@
...
+    @NotNull
     private String name;
 
+    @Temporal(TemporalType.DATE)
+    @NotNull
     private LocalDateTime birthDate;
 
+    @ElementCollection
+    @NotNull
+    private Set<LanguageEnum> languages = new HashSet<>();
```
[the remark this answers is not in the document]
Comment (adamkral12) [reply Secret-agency-persistence/src/main/java/cz/fi/muni/pa165/secretagency/entity/Agent.java]: jj, melo by to jit, pridal jsem tam taky enumerated

**Comment by adamkral12** [comment]
updated

**Pull request merged**